tests/integration/app-charm/src/charm.py

```diff
@@ -242,6 +275,51 @@ def _http_request(self, event) -> None:
         response = http_request(credentials, endpoint, method, payload)
         event.set_results({"results": json.dumps(response)})
 
+    def _generate_cert(self, event) -> None:
+        """Handle the action that generates a certificate."""
+        if not self.unit.is_leader():
+            event.fail("The action can be run only on leader unit.")
+            return
+
+        # read parameters from the event
+        common_name: str = event.params["common-name"]
+
+        # generate the certificate
+        cert, key = generate_cert(common_name)
+        # save the certificate and key to disk
+        etcd_snap_dir = Path(ETCD_SNAP_DIR)
+        etcd_snap_dir.mkdir(parents=True, exist_ok=True)
+        Path(etcd_snap_dir / "client.pem").write_text(cert)
+        Path(etcd_snap_dir / "client.key").write_text(key)
```

Comment:
Could you put these in a directory `certificates` or similar instead of putting them in the root of SNAP_COMMON?